src/utils/settings.cpp

```diff
@@ -74,9 +74,13 @@ std::unique_ptr<QSettings> persistent_settings(const QString& key)
 
 bool exists_but_unreadable(const QString& filename)
 {
-    QFile file(filename);
-    return !file.open(QIODevice::ReadOnly) && errno != ENOENT; /* note file.error() not enough for us: it would not
-                                                                  distinguish the cause of failure */
+    std::ifstream stream;
+    stream.open(filename.toStdString(), std::ios_base::in);
+    return stream.fail() && errno && errno != ENOENT; /*
+        Note: file.error() is not enough for us: it would not distinguish the cause of failure;
```

Comment:
I'm not sure what `file.error()` is referring to in these comments.

Reply:
Indeed, `file` went away now, will have to update it, thanks.

Reply:
Done, I used QFile::error - although `QFile` isn't used there anymore, `QFile::error` would be a suitable alternative if it gave us the same info as `QFile::errorString`. If you prefer I can remove the comment entirely.

Reply:
I think leaving the comment for now is fine for historical purposes.